scripts/test_updates.sh

```diff
@@ -24,33 +27,42 @@ exit 1
 
-docker run -d --name timescaledb-clean -v /tmp/pg_data_clean:/var/lib/postgresql/data -p ${CLEAN_PG_PORT}:5432 update_test:latest
-docker run -d --name timescaledb-orig -v ${PGTEST_TMPDIR}/pg_data:/var/lib/postgresql/data -p ${UPDATE_PG_PORT}:5432 timescale/timescaledb:${UPDATE_FROM_TAG}
+docker run -d --name timescaledb-orig -v ${UPDATED_DATA_DIR}:/var/lib/postgresql/data -p ${UPDATE_PG_PORT}:5432 timescale/timescaledb:${UPDATE_FROM_TAG}
+docker run -d --name timescaledb-clean -v ${CLEAN_DATA_DIR}:/var/lib/postgresql/data -p ${CLEAN_PG_PORT}:5432 update_test:latest
 
 wait_for_pg ${UPDATE_PG_PORT}
 
+echo "Executing setup script on 0.1.0"
 psql -h localhost -U postgres -p ${UPDATE_PG_PORT} -f test/sql/updates/setup.sql
 docker rm -vf timescaledb-orig
 
-docker run -d --name timescaledb-updated -v /tmp/pg_data:/var/lib/postgresql/data -p ${UPDATE_PG_PORT}:5432 update_test:latest
+docker run -d --name timescaledb-updated -v ${UPDATED_DATA_DIR}:/var/lib/postgresql/data -p ${UPDATE_PG_PORT}:5432 update_test:latest
 
 wait_for_pg ${UPDATE_PG_PORT}
 
+echo "Executing ALTER EXTENSION timescaledb UPDATE"
 psql -h localhost -U postgres -d single -p ${UPDATE_PG_PORT} -c "ALTER EXTENSION timescaledb UPDATE"
 
 
 wait_for_pg ${CLEAN_PG_PORT}
 
+echo "Executing setup script on new version"
 psql -h localhost -U postgres -p ${CLEAN_PG_PORT} -f test/sql/updates/setup.sql
-psql -h localhost -U postgres -d single -p ${UPDATE_PG_PORT} -f test/sql/updates/test-0.1.1.sql > /tmp/updated.out
-psql -h localhost -U postgres -d single -p ${CLEAN_PG_PORT} -f test/sql/updates/test-0.1.1.sql > /tmp/clean.out
 
-docker rm -f timescaledb-updated timescaledb-clean || rm -rf  ${PGTEST_TMPDIR}/pg_data ${PGTEST_TMPDIR}/pg_data_clean
+echo "Restarting clean container"
+#below is needed so the clean container looks like updated, which has been restarted after the setup script
+#(especially needed for sequences which might otherwise be in different states -- e.g. some backends may have reserved batches)
+docker rm -vf timescaledb-clean
+docker run -d --name timescaledb-clean -v ${CLEAN_DATA_DIR}:/var/lib/postgresql/data -p ${CLEAN_PG_PORT}:5432 update_test:latest
+wait_for_pg ${CLEAN_PG_PORT}
+
+echo "Testing"
+psql -X -v ECHO=ALL -h localhost -U postgres -d single -p ${UPDATE_PG_PORT} -f test/sql/updates/test-0.1.1.sql > ${PGTEST_TMPDIR}/updated.out
+psql -X -v ECHO=ALL -h localhost -U postgres -d single -p ${CLEAN_PG_PORT} -f test/sql/updates/test-0.1.1.sql > ${PGTEST_TMPDIR}/clean.out
 
-diff ${PGTEST_TMPDIR}/clean.out ${PGTEST_TMPDIR}/updated.out > ${PGTEST_TMPDIR}/update_test.output
-cat ${PGTEST_TMPDIR}/update_test.output
+docker rm -f timescaledb-updated timescaledb-clean || rm -rf  ${CLEAN_DATA_DIR} ${UPDATED_DATA_DIR}
 
-cmp ${PGTEST_TMPDIR}/clean.out ${PGTEST_TMPDIR}/updated.out > ${PGTEST_TMPDIR}/update_test.output
+diff ${PGTEST_TMPDIR}/clean.out ${PGTEST_TMPDIR}/updated.out | tee ${PGTEST_TMPDIR}/update_test.output
```

Comment:
I think this is wrong. The tee will make it return 0 even if there is a diff right ? That is why it used to be a cmp.